.github/workflows/test.yml

```
@@ -1,35 +1,44 @@
+          fi
+          echo "✅ Source branch validation passed"
+  test:
+    runs-on: macos-latest
+    needs: validate
+    if: github.head_ref == 'dev'
 
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v4
+      - name: Check out Git repository
+        uses: actions/checkout@v3
```

actionlint (1.7.10)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)